src/domain.py

```diff
@@ -109,6 +110,9 @@ def add_scrape_config(self, scrape_config):
     def yaml_dump(self):
         return yaml.dump(self._config_dict)
 
+    def dump(self):
+        return self._config_dict
+
```

Comment:
The method name is misleading since `dump` usually returns a string. Maybe you can make this more consistent with other classes that use `to_dict` as the method name?